src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -248,23 +259,43 @@ public void Reset()
             _abortedCts = null;
         }
 
-        protected void ResetComponents()
+        protected void ResetComponents(bool immediateReuse)
         {
-            var frameHeaders = Interlocked.Exchange(ref _frameHeaders, null);
-            if (frameHeaders != null)
+            if (immediateReuse && HeaderFactory != null)
```

Comment:
Yeah. So now I see we *are* still caching, but this way of configuring it seems bad.